massa-execution-worker/src/execution.rs

```diff
@@ -445,6 +449,57 @@ impl ExecutionState {
             )));
         }
 
+        // Check selector
+        // Note 1: Has to be done after slot limit but before executed check
+        // Note 2: that this is done for a node to create a Block with 'fake' denunciation thus
+        //       include them in executed denunciation and prevent (by occupying the corresponding entry)
+        //       any further 'real' denunciation.
+
+        match &denunciation {
+            Denunciation::Endorsement(_de) => {
+                // Get selected address from selector and check
+                let selected = self.selector.get_selection(*de_slot);
+                match selected {
+                    Ok(selection) => {
+                        if let Some(address) = selection
+                            .endorsements
+                            .get(*denunciation.get_index().unwrap_or(&0) as usize)
+                        {
+                            if *address != addr_denounced {
+                                return Err(ExecutionError::IncludeDenunciationError(
+                                    "Attempt to execute a denunciation but address was not selected".to_string()
+                                ));
+                            }
+                        } else {
+                            // Panic here as this should never happen
+                            panic!("Could not get selected address for endorsements at index");
```

Comment:
if we panic here, we need to make sure that denunciation deserialization fails if index >= ENDORSEMENT_COUNT, otherwise it's an attack vector

Reply:
Note: the code can be simplified:

```
let selected_addr = self
    .selector
    .get_selection(*de_slot)
    .expect("could not get selection blablabla")
    .endorsements
    .get(_de.index)
    .expect("could not get selection at index blablabla");  // TODO ensure that the denunciationdeserializer prevents this from happening 
```

Reply:
I've checked it and the deserializer is bound by ENDORSEMENT_COUNT

Reply:
Code simplified